### DISCUSSION TITLE
Can I develop custom markdown render formats?

### DISCUSSION THREAD

**Discussion by MichelNivard** [quarto-dev/quarto-cli#11621] [Q&A]
### Description

I see there are quarto extensions for specific formats, their allt typist/html/pdf is it possible to author extensions to output custom markdown formats?

**Comment by mcanouil** [comment]
I believe, you can extend all the outputs formats available in Quarto but it will likely be through Lua filters or post-render script while Typst, LaTeX, and HTML can use partials/templates.

**Comment by dcnorris** [comment]
Could you offer more guidance on the architectural thinking behind excluding Markdown formats from this mode of extension?  Is there some reason why a custom format based on one of Pandoc's [many Markdown output formats](https://pandoc.org/MANUAL.html#option--to) would constitute a poor design?
```bash
$ quarto create extension format:commonmark
? Extension Name › PubPeer
ERROR: The sub type commonmark isn't valid. Expected one of html, pdf, docx, revealjs, typst
```

My thought was to develop a `PubPeer` format (deriving from `commonmark_x-smart` as that gets pretty close to the desired output) incorporating a custom Lua filter to trim leftovers that PubPeer's [rather austere Markdown](https://pubpeer.com/static/markdown) won't accept.  From the Quarto user's perspective (mine!), it would be convenient to develop a comment in an HTML format, but then switch to `format: PubPeer` once it was ready to post.  But from what I gather thus far, it seems a [custom Markdown writer](https://pandoc.org/custom-writers.html#example-modified-markdown-writer) might be the best-supported approach.

**Reply by cscheid** [reply]
> Could you offer more guidance on the architectural thinking behind excluding Markdown formats from this mode of extension?

I think the answer is simple and not very satisfying: not enough hours in the day between the developers.

**Reply by dcnorris** [reply]
That's actually very helpful, thanks!
